llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -6947,6 +6947,30 @@ bool llvm::onlyUsedByLifetimeMarkersOrDroppableInsts(const Value *V) {
       V, /* AllowLifetime */ true, /* AllowDroppable */ true);
 }
 
+bool llvm::isLanewiseOperation(const Instruction *I) {
+  if (auto *II = dyn_cast<IntrinsicInst>(I)) {
```

Comment:
```
  if (isBinaryOp() || isUnaryOp())
    return true;
  if (isCast() && !isa<BitCastInst>(this))
    return true;
  if (isa<SelectInst>(this) || isa<CmpInst>(this))
    return true;
```
probably

Reply:
Redundant? We just check-negative on lane-crossing instructions.

Reply:
Yeah, I was thinking you would do too helpers, but this works.